### PR TITLE
feat: add C implementation for `math/base/special/fast/min`

### LINKED ISSUE #1911: [RFC]: add C implementation for `math/base/special/fast/min`
### Description

This RFC proposes add C implementation for `math/base/special/fast/min`

### Related Issues

Related issues #649  .

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Userf1995**
Resolves #1911  .

## Description

> What is the purpose of this pull request?

This pull request:

-   add C implementation for math/base/special/fast/min

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1911 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [X] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/fast/min/examples/c/example.c [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    double y;
+    double v;
+    int i;
+    
+    for ( i = 0; i < 100; i++ ) {
+        x = ( ( (double)rand() / (double)RAND_MAX ) * 10000.0 ) - 5000.0;
+        y = ( ( (double)rand() / (double)RAND_MAX ) * 10000.0 ) - 5000.0;
+        v = stdlib_base_fast_min( x, y );
+        printf( "x: %lf, y: %lf, min(x, y): %lf\n", x, y, v );
+    }
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/fast/min/examples/c/example.c]:
```suggestion
	double x;
	double y;
	double v;
	int i;
    
	for ( i = 0; i < 100; i++ ) {
		x = ( ( (double)rand() / (double)RAND_MAX ) * 10000.0 ) - 5000.0;
		y = ( ( (double)rand() / (double)RAND_MAX ) * 10000.0 ) - 5000.0;
		v = stdlib_base_fast_min( x, y );
		printf( "x: %lf, y: %lf, min(x, y): %lf\n", x, y, v );
	}
```

File: lib/node_modules/@stdlib/math/base/special/fast/min/lib/native.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+// MODULES //
+
+var addon = require( './../src/addon.node' );
+
+
+// MAIN //
+
+/**
+* Return the minimum value.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/fast/min/lib/native.js]:
```suggestion
* Returns the minimum value.
```

File: lib/node_modules/@stdlib/math/base/special/fast/min/manifest.json [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+      ],
+      "include": [
+        "./include"
+      ],
+      "libraries": [],
+      "libpath": [],
+      "dependencies": []
+    }
+  ]
+}
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/fast/min/manifest.json]:
```suggestion
}

```

File: lib/node_modules/@stdlib/math/base/special/fast/min/src/main.c [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+*
+* @example
+* double v = stdlib_base_fast_min( 0.0, -0.0 );
+* // returns -0.0
+*/
+double stdlib_base_fast_min( const double x, const double y ) {
+    if ( x < y ) {
+        return x;
+    }
+    return y;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/fast/min/src/main.c]:
```suggestion
	if ( x < y ) {
		return x;
	}
	return y;
```

**Review by Pranavchiku — APPROVED** [review]
Looks good. Thanks @Rejoan-Sardar !

File: lib/node_modules/@stdlib/math/base/special/fast/min/lib/native.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+* var v = min( 0.0, -0.0 );
+* // returns -0.0
+*
+* @example
+* var v = min( 3.14, NaN );
+* // returns NaN
+*
+* @example
+* var v = min( NaN, 5.0 );
+* // returns NaN
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/fast/min/lib/native.js]: Per tests in `test.native.js`, this should be:

```suggestion
* var v = min( NaN, 5.0 );
* // returns 5.0
```

**Review by Planeshifter — APPROVED** [review]

**Pull request merged**